app/models/institution_builder.rb

```diff
@@ -603,22 +643,27 @@ def self.build_versioned_school_certifying_official(version_number)
         phone_number,
         phone_extension,
         email,
-        version)
-      Select
-        facility_code,
-        institution_name,
-        priority,
-        first_name,
-        last_name,
-        title,
-        phone_area_code,
-        phone_number,
-        phone_extension,
-        email,
-        ?
-      FROM school_certifying_officials
+        i.id
+      FROM school_certifying_officials s
```

Comment:
```
INSERT INTO school_certifying_officials
...
SELECT DISTINCT
...
FROM school_certifying_officials
```
Insert records into a table from itself?

Reply:
That is a bit clunky, but it seemed to accomplish what I was going for at the time.

I refactored it a little bit to what I have below in the latest commit, which seems a bit less clunky.
```
INSERT INTO school_certifying_officials(
          facility_code,
          institution_name,
          priority,
          first_name,
          last_name,
          title,
          phone_area_code,
          phone_number,
          phone_extension,
          email,
          institution_id)
       SELECT
        i.facility_code,
        i.institution,
        s.priority,
        s.first_name,
        s.last_name,
        s.title,
        s.phone_area_code,
        s.phone_number,
        s.phone_extension,
        s.email,
        i.id
      FROM institutions i
      INNER JOIN school_certifying_officials s ON i.facility_code = s.facility_code
      INNER JOIN versions v ON v.number = ?
      WHERE v.id = i.version_id
      AND i.id IS NOT NULL;
```

Reply:
OK, let me step through the life of this table.

I upload a `sco.csv` with two records:
![sco_start](https://user-images.githubusercontent.com/733873/74353781-67c3d600-4d88-11ea-9772-a37553b26b3b.png)

Now I generate a new preview version and check my table again:
![sco_run1](https://user-images.githubusercontent.com/733873/74353876-87f39500-4d88-11ea-98e4-2c0d1c9271bf.png)
What happened?
`build_school_certifying_officials` added two new records with the `institution_id` populated as expected.  Then `add_sco_institution_id` ran and updated the original two records with an `institution_id`, so now we have duplicate data.

Then I generate another preview version, and check my table again.
![sco_run2](https://user-images.githubusercontent.com/733873/74354276-1c5df780-4d89-11ea-953b-370b5f6916fb.png)
Instead of adding two new rows for the new version, it doubled the number of rows.
What happened?
Your SQL joins `school_certifying_officials` to `institutions` on `facility_code`.  All 4 existing records join on that condition and thus 4 new records were inserted instead of the expected 4.

Finally, I upload my`sco.csv` file again.
![re-up](https://user-images.githubusercontent.com/733873/74354526-7fe82500-4d89-11ea-8c84-c6c99c1ec857.png)

Whoops!  When you upload a file it deletes all the existing data before importing the new data.  Looks like your still going to need a second table.

Reply:
Hi @jholton, 

I have actually run into this duplication issue before. I believe when I updated the query yesterday I dropped off the `UNIQUE` keyword from the query which is how I believe I had resolved this in the past. I will update the query adding that back in to see if resolves this duplication issue.

Reply:
Don't miss my last point.  Fixing the duplication won't matter if the table gets wiped out when someone uploads a new csv.  😃

Reply:
Hey @jholton,

My apologies, I meant to loop back with you here. That point was not lost, I just pushed that commit correcting the duplication issue for testing purposes.

After discussing a plan forward with some of my team members I am working on re-implementing that functionality to the second table `versioned_school_certifying_officials`.

I saw exactly the behavior you described earlier and am working on a fix now 👍

Reply:
Sounds good; just wanted to make sure.  I think a second table is the way to go.

Reply:
Pushed an update that reconfigures `versioned_school_certifying_officials` to work with the refactored archiver.

Reply:
That probably makes sense to go ahead and do, I'll add that migration to do so.